android/java/org/chromium/chrome/browser/search_engines/settings/BraveSearchEngineAdapter.java

```diff
@@ -26,46 +27,66 @@ public class BraveSearchEngineAdapter extends SearchEngineAdapter {
     public static final String STANDARD_DSE_SHORTNAME = "standard_dse_shortname";
 
     private boolean mIsPrivate;
+    private Profile mProfile;
 
-    static public void setDSEPrefs(TemplateUrl templateUrl, boolean isPrivate) {
+    public BraveSearchEngineAdapter(Context context, boolean isPrivate) {
+        super(context);
+        mIsPrivate = isPrivate;
+
+        // Only need last used profile because we are in settings
+        if (mProfile == null) {
+            if (!mIsPrivate) {
+                mProfile = Profile.getLastUsedRegularProfile().getOriginalProfile();
+            } else {
+                mProfile = Profile.getLastUsedRegularProfile().getPrimaryOTRProfile(/* createIfNeeded= */ true);
+            }
+        }
+    }
+
+    static public void setDSEPrefs(TemplateUrl templateUrl, Profile profile) {
+        Log.d("BraveSearchEngineAdapter", "setDSEPrefs: isPrivate: " + String.valueOf(profile.isOffTheRecord()) + " ShortName: " + templateUrl.getShortName());
```

Comment:
do we need logs?

Reply:
I'll do a clean up after fixed other problems. Same for the other logs.